packages/ceramic-core/src/anchor/memory/in-memory-anchor-service.ts

```diff
@@ -94,43 +121,64 @@ class InMemoryAnchorService extends AnchorService {
     this._dispatcher = this._ceramic.dispatcher
   }
 
-  async requestAnchor(docId: string, head: CID): Promise<void> {
-    const pair: CidDoc = {
-      docId, cid: head,
-    }
+  /**
+   * Send request to the anchoring service
+   * @param docId - Document ID
+   * @param cid - Record CID
+   */
+  async requestAnchor(docId: string, cid: CID): Promise<void> {
+    const candidate: Candidate = new Candidate(cid, docId)
 
     if (this._anchorOnRequest) {
-      await this._process(pair)
+      await this._process(candidate)
     } else {
-      this._queue.push(pair)
+      this._queue.push(candidate)
     }
   }
 
   /**
-   * Process single pair
-   * @param cidDoc - Document-CID pair
+   * Process single candidate
    * @private
    */
-  async _process(cidDoc: CidDoc): Promise<void> {
+  async _process(leaf: Candidate): Promise<void> {
     // creates fake anchor record
     const proofData: AnchorProof = {
       chainId: 'eip155:1',
       blockNumber: Date.now(),
       blockTimestamp: Date.now(),
       txHash: new CID(this.SAMPLE_ETH_TX_HASH),
-      root: cidDoc.cid,
+      root: leaf.cid,
     }
     const proof = await this._dispatcher.storeRecord(proofData)
-    const record = { proof, path: '', prev: cidDoc.cid }
+    const record = { proof, path: '', prev: leaf.cid }
     const cid = await this._dispatcher.storeRecord(record)
 
     // add a delay
     const handle = setTimeout(() => {
-      this.emit(cidDoc.docId, { status: 'COMPLETED', message: 'CID successfully anchored.', anchorRecord: cid });
+      this.emit(leaf.docId, { status: 'COMPLETED', message: 'CID successfully anchored.', anchorRecord: cid })
       clearTimeout(handle)
     }, this._anchorDelay)
   }
 
+  /**
+   * Verifies record signature
+   * @param record - Record data
+   * @return DID
+   * @private
+   */
+  async verifySignedRecord(record: any): Promise<string> {
+    const { payload, signatures } = record
+    const { signature, protected: _protected } = signatures[0]
+
+    const decodedHeader = JSON.parse(base64url.decode(_protected))
+    const { kid } = decodedHeader
+
+    const didDoc = await this._ceramic.context.resolver.resolve(kid)
+    const jws = [_protected, payload, signature].join(".")
+    await didJwt.verifyJWS(jws, didDoc.publicKey)
+    return kid
```

Comment:
We likely need to strip the extra data from the DID. Should only be `did:3:bafy...`. Should not include `version-id` or `#<keyfragment`.

Reply:
yeah, I did that initially but I changed my mind :) I'll revert it, tnx!